src/stores/blockstore.js

```diff
@@ -62,28 +69,50 @@ exports.setUp = (basePath, BlobStore, locks) => {
       }
 
       const blockPath = multihashToPath(key)
-      const deferred = pullDefer()
+      const deferredSource = pullDefer.source()
+      // const deferredSource = pullDefer.source()
 
+      // lock, read from all stores in parallel, take first returned result, and release
       lock(blockPath, (release) => {
-        pull(
-          store.read(blockPath),
-          pull.collect(release(released))
-        )
+        const onComplete = release(onReleased)
+        let didComplete = false
+        each(stores, (store, cb) => {
```

Comment:
Something like `race` would make more sense here, as you only care for the first one to return.